SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriverRPC.h

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Interop/RPCs/RPCService.h"
+#include "SpatialView/EntityComponentId.h"
+#include "SpatialNetDriverRPC.generated.h"
+
+namespace SpatialGDK
+{
+class SpatialEventTracer;
+}
+
+struct FRPCPayload
+{
+	uint32 Offset;
+	uint32 Index;
+	TArray<uint8> PayloadData;
+
+	void ReadFromSchema(Schema_Object* RPCObject);
+	void WriteToSchema(Schema_Object* RPCObject) const;
```

Comment:
is this const the right way around? I would except read to be const and write not, or something else going on here?

Reply:
ReadFromSchema takes data from RPCObject to put into this
WriteToSchema copies data from this to put into RPCObject.

Reply:
So can RPCObject be const for ReadFromSchema?

Reply:
Just realized the worker SDK APIs are const-correct.

Reply:
Names could be improved for clarity.
"ReadFromSchema" should take a const ref.
"WriteToSchema" should take a ref.

Reply:
Regarding using ref for schema types, the worker SDK uses raw pointers, and we use raw pointers for these types throughout the GDK.
I would rather keep that pattern unless we uniformize our usage of these types throughout the GDK.

Reply:
I don't like the pattern personally, but happy to review at a later date.